components/producers/semgrep/types/semgrep-issue.go

```diff
@@ -33,4 +33,6 @@ type SemgrepResults struct {
 type Metavars struct{}
 
 // Metadata currently is empty, however, could represent semgrep issue metadata going forward.
-type Metadata struct{}
+type Metadata struct {
+	CWE interface{} `json:"cwe"` // Can be a []string but also just a string
```

Comment:
why is this handled in a separate function and we don't override instead the JSON marshaller so that we can have a CWE []int instead of having an `interface{}`

Reply:
No idea how that would work or what that would look like, tbh - can u give me any pointers?

Reply:
https://pkg.go.dev/encoding/json#Marshaler

Reply:
so you mean:

1. change the `Metadata.CWE: int[]`
2. in the `main`, do a
```diff
- if err := json.Unmarshal(inFile, &results); err != nil {
+ if err := ourCustomJSONUnmarshalThatHandlesTheCWE(...`
```

where
```golang
func ourCustomJSONUnmarshalThatHandlesTheCWE(...) {
  json.Unmarshal(...)
  // also handle the interface{} -> int[] conversion
}
```

is that what you mean?

Reply:
Yes we should use properly types fields unless there is a very good reason not to. I have rarely seen a good reason to use the `interface{}`.
No, that's not what the docs of the json library say. There are plenty of examples of how you implement custom JSON marshaling/unmarshaling in Golang: https://gobyexample.com/json.